src/jvmTest/kotlin/it/krzeminski/githubactionstyping/JsonSchemaValidatorSchemaValidationTest.kt

```diff
@@ -0,0 +1,134 @@
+package it.krzeminski.githubactionstyping
+
+import io.github.optimumcode.json.schema.JsonSchema
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.datatest.withData
+import io.kotest.matchers.Matcher
+import io.kotest.matchers.MatcherResult
+import io.kotest.matchers.file.shouldBeADirectory
+import io.kotest.matchers.file.shouldNotBeEmptyDirectory
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldNot
+import io.kotest.mpp.sysprop
+import it.krzeminski.snakeyaml.engine.kmp.api.Load
+import kotlinx.serialization.json.JsonArray
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonNull
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import java.io.File
+
+private val schemaFile = File(sysprop("schemaFile")!!)
+private val catalogDir = File(sysprop("catalogDir")!!)
+private val externalDir = File(sysprop("externalDir")!!)
+private val goodDir = File(sysprop("goodDir")!!)
+private val badDir = File(sysprop("badDir")!!)
+
+private lateinit var schema: JsonSchema
+
+/**
+ * Validate good and bad typings against the Json Schema Validator schema validator.
+ */
+class JsonSchemaValidatorSchemaValidationTest : FunSpec({
```

Comment:
It would be cool to check if the current action's logic reports a problem with the typing whenever the schema validation fails + the success cases as well. It looks like a topic for a separate set of changes, because I presume there may be some minor discrepancies that we should consider carefully case by case.

Reply:
I agree that such a check makes sense, then you are also sure that for each violation you get a nice error message and that you do not get one if there is no validation.

In that test it can also be checked that for the violations the according error message is shown to be meaningfully correct. But I also think that this should be a separate work-item.

Reply:
https://github.com/typesafegithub/github-actions-typing/issues/253